frontend/server/src/main/java/org/pytorch/serve/wlm/ModelManager.java

```diff
@@ -205,87 +204,139 @@ private ModelArchive createModelArchive(
         return archive;
     }
 
+    private void setupModelVenv(Model model)
+            throws IOException, InterruptedException, ModelException {
+        String venvPath = EnvironmentUtils.getPythonVenvPath(model);
+        List<String> commandParts = new ArrayList<>();
+        commandParts.add(EnvironmentUtils.getPythonRunTime(model));
+        commandParts.add("-m");
+        commandParts.add("venv");
+        commandParts.add("--clear");
```

Comment:
why clear? Seems beneficial to allow users to install their dependencies beforehand 

It was always quite weird how we pip installed a bunch of stuff on launching a model

Reply:
Sounds good, will check if `venvs` can safely be made portable, since the symlinks in the `bin` directory of the venv, for ex: `venv/bin/python` may need to be updated for them to work, since the python binary path may not be the same on the host on with the venv is created and the host on which the venv is used.

From the official docs: https://docs.python.org/3/library/venv.html
```
Warning: Because scripts installed in environments should not expect the environment to be activated, their shebang lines contain the absolute paths to their environment’s interpreters. Because of this, environments are inherently non-portable, in the general case. You should always have a simple means of recreating an environment (for example, if you have a requirements file requirements.txt, you can invoke pip install -r requirements.txt using the environment’s pip to install all of the packages needed by the environment).
```